crypto/bytestring/cbs.c

```diff
@@ -520,7 +520,9 @@ int CBS_get_asn1_int64(CBS *cbs, int64_t *out) {
   }
   uint8_t sign_extend[sizeof(int64_t)];
   memset(sign_extend, is_negative ? 0xff : 0, sizeof(sign_extend));
-  for (size_t i = 0; i < len; i++) {
+  // GCC 12/13 report `stringop-overflow` on the following line
+  // without additional condition: `i < sizeof(int64_t)`
+  for (size_t i = 0; i < len && i < sizeof(int64_t); i++) {
```

Comment:
Yes, I believe so. But `gcc-13` seems to think otherwise: 
```
/Users/justsmth/repos/aws-lc/crypto/bytestring/cbs.c: In function 'CBS_get_asn1_int64':
/Users/justsmth/repos/aws-lc/crypto/bytestring/cbs.c:534:20: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
  534 |     sign_extend[i] = data[len - i - 1];
      |     ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
/Users/justsmth/repos/aws-lc/crypto/bytestring/cbs.c:521:11: note: at offset 8 into destination object 'sign_extend' of size 8
  521 |   uint8_t sign_extend[sizeof(int64_t)];
      |           ^~~~~~~~~~~

¯\_(ツ)_/¯
```